packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -2869,6 +2895,13 @@ class SemanticsNode with DiagnosticableTreeMixin {
   /// {@endtemplate}
   SemanticsRole get role => _role;
   SemanticsRole _role = _kEmptyConfig.role;
+  set role(SemanticsRole value) {
```

Comment:
we should avoid this

Reply:
This is removed after offline discussion.

we won't directly change semantics node's role, just wrap them with a new semantics node with a cell role.